health-services/ExerciseSample/app/src/main/java/com/example/exercise/HealthServicesManager.kt

```diff
@@ -218,7 +218,11 @@ class HealthServicesManager @Inject constructor(
         }
         exerciseClient.setUpdateCallback(callback)
         awaitClose {
-            exerciseClient.clearUpdateCallbackAsync(callback)
+            runBlocking {
+                launch {
+                    exerciseClient.clearUpdateCallback(callback)
+                }
+            }
```

Comment:
Not 100% sure if this is the right thing to do here. Doesn't compile without the `runBlocking { launch { ... } }` wrapper.

Reply:
How about just : 

```
runBlocking {
    exerciseClient.clearUpdateCallback(callback)
}
```

I don't think you want to launch another coroutine in the `runBlocking` block?

Reply:
🤦 Thanks, fixed!